.github/workflows/e2e-pr.yml

```diff
@@ -0,0 +1,112 @@
+name: SalesforceCommerceCloud/pwa-kit/e2e-pr
+on:
+    workflow_dispatch:
+    pull_request: # Default: opened, reopened, synchronize (head branch updated)
+    merge_group: # Trigger GA workflow when a pull request is added to a merge queue.
+    push:
+        branches:
+            - develop
+            - 'release-*'
+
+jobs:
+    test_e2e_private:
+        runs-on: ubuntu-latest
+        steps:
+           # Skipping the entire workflow for now until all steps are implemented.
+            - name: Skip Check
+              run: | 
+                echo "SKIP_WORKFLOW=true" >> "$GITHUB_ENV"
+
+            - name: Checkout
+              uses: actions/checkout@v4
+
+            - name: Check PWA Kit Version
+              run: |-
+                  version=`jq -r ".version" package.json`
+                  echo "pwa_kit_version=$version" >> "$GITHUB_ENV"
+
+            # TODO: Skip the entire workflow since we don't have e2e tests for PWA Kit v2.x
+            - name: Skip if PWA Kit version older than v3.x
+              if: ${{ env.SKIP_WORKFLOW != 'true' }}
+              run: |
+                  major_version=$(echo "${{ env.pwa_kit_version }}" | cut -d. -f1)
+                   if [ "$major_version" -lt 3 ]; then
+                     echo "PWA Kit version is older than v3.x, skipping workflow."
+                     echo "SKIP_WORKFLOW=true" >> "$GITHUB_ENV"
+                   fi
```

Comment:
nit: Would it be useful to separate these Pre-E2E tests steps in a separate job to track progress in the GA builds and handle failures better?

Reply:
Ideally yes. But I left them in the same job because if the test run ends the environment will be released back to the pool. (Env won't be destroyed but it'll be made available for some other PR to pick up)

So acquiring the lock or getting a MRT env is a part of the test run and must happen each time the tests are executed. So I left those steps in the same job.